static/app/components/searchQueryBuilder/tokens/filter/valueListBox.tsx

```diff
@@ -55,13 +92,43 @@ export function ValueListBox<T extends SelectOptionOrSectionWithKey<string>>({
   items,
   canUseWildcard,
   portalTarget,
+  wrapperRef,
 }: ValueListBoxProps<T>) {
   const totalOptions = items.reduce(
     (acc, item) => acc + (itemIsSection(item) ? item.options.length : 1),
     0
   );
   const anyItemsShowing = totalOptions > hiddenOptions.size;
 
+  const listBoxRefCallback = useCallback((element: HTMLUListElement | null) => {
+    listBoxRef.current = element;
+
+    if (!element) return undefined;
+
+    const refsToSync = [listBoxRef, popoverRef];
+
+    constrainAndAlignListBox({
+      popoverRef,
+      refsToSync,
+      referenceRef: wrapperRef,
+    });
+
+    const observer = new ResizeObserver(() => {
+      constrainAndAlignListBox({
+        popoverRef,
+        refsToSync,
+        referenceRef: wrapperRef,
+      });
+    });
+
+    observer.observe(element);
+
+    return () => {
+      observer.disconnect();
+    };
+    // all deps are refs but because they're props, the linter complains
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
if they are refs, then there’s no harm in adding them to the dependency array, as they won’t change during render.

The advantage is that if this function evolves and gets real dependencies, we will be forced to add them. with a disable, they can silently creep in.

Reply:
Ahh okay, sounds good!